torch/testing/_internal/common_methods_invocations.py

```diff
@@ -655,7 +655,7 @@ def __init__(self,
 
         self.decorators = (*decorators, *skips)
 
-        self.sample_inputs_func = sample_inputs_func
+        self.sample_inputs_func = torch.no_grad()(sample_inputs_func)
```

Comment:
nb. This is the important point of this PR. The rest of the PR just rewrites some tests for them to work as expected with this change and with the `non_contiguous` tests added in https://github.com/pytorch/pytorch/pull/67996

Reply:
Comment here too